nfmodel/config.txt
--nohelp
--nohelpfull
--nohelpshort
--nohelpxml
--noonly_check_args
--nopdb
--nopdb_post_mortem
--norun_with_pdb
--norun_with_profiling
--use_cprofile_for_profiling
--noalsologtostderr
--log_dir=
--logger_levels=
--nologtostderr
--showprefixforinfo
--stderrthreshold=fatal
--verbosity=0
--accumulate=1
--anneal_method=cosine
--anneal_point=0.5
--atten_factor=3.0
--backbone=gcn_equi4
--batch_size=4
--big_in_whole=1
--dataset=Real
--dataset_dir=/home/bingxing2/home/scx6md6/wby/nocs_data
--dataset_size=big
--debug_path=/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/debug.pt
--detection_dir=/data_nvme/nocs_data/segmentation_results
--device=cuda:0
--dim_list=128,128,256,256,512
--down_sample_method=farthest
--equi_neighbor_num=10
--eval_dataset=Real
--eval_out=output/pips_equi_v5
--fix_kernel=0
--gamma=0.1
--log_every=10
--lr=0.0003
--lr_decay_iters=50
--lr_pose=1.0
--lr_scheduler_name=flat_and_anneal
--model_save=output/pips_equi4_tovec3_no_global_pool
--nu_nocs=0
--num_workers=4
--obj_c=6
--optimizer_type=Ranger
--per_obj=mug
--pick_ratio=0.0
--poly_power=0.9
--qnet_config=qnet_equi_512.yaml
--qnet_version=v5
--random_points=1028
--resume=0
--resume_dir=/data_nvme/trained_models/pips_equi3_tovec3_no_global
--resume_model_name=model_49.pth
--resume_point=69
--rotation_noise_angle=20.0
--rotation_path=/home/bingxing2/home/scx6md6/wby/SOCS/nfmodel/rotation.pt
--sample_method=farthest
--save_every=5
--scale_invariant=0
--shuffle=1
--softmax=1
--total_epoch=70
--tovec_version=v3
--train=1
--train_steps=2000
--use_board=1
--use_global=0
--use_pick=0
--use_pool=0
--use_pred_scale=0
--verbose=0
--warmup_factor=0.001
--warmup_iters=1000
--warmup_method=linear
--weight_decay=0.0
--DZI_PAD_SCALE=1.5
--DZI_SCALE_RATIO=0.25
--DZI_SHIFT_RATIO=0.25
--DZI_TYPE=uniform
--R_c=3
--Ts_c=6
--align_origin=0
--aug_bb_pro=0.3
--aug_bc_pro=0.3
--aug_bg=1
--aug_deform=0
--aug_pc_pro=1.0
--aug_pc_r=0.001
--aug_rt_pro=1.0
--base_dir=/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/output/NFwhole_prior_7
--bin_size=100
--cd_w=0.0
--consistency_w=1.0
--cur_cat_model_num=0
--cur_eval_index=-1
--cut_backbone=0
--debug=0
--deform_w=10.0
--eval_inference_only=0
--eval_optimizer=adam
--eval_use_deform=0
--eval_visualize_pcl=0
--face_recon_c=30
--fake_grid_num=500
--fake_radius=0.6
--feat_c_R=1280
--feat_c_ts=1283
--feat_face=768
--feat_for_ts=0
--force_sym=0
--fsnet_loss_type=l1
--gcn_n_num=10
--gcn_sup_num=7
--grad_w=1.0
--img_size=256
--inter_R_lr=1.0
--inter_nocs_lr=1.0
--inter_t_lr=1.0
--interpo_w=10.0
--keypoint_num=36
--keypoint_path=/home/bingxing2/home/scx6md6/wby/nocs_data/keypoint.pkl
--kld_w=0.0
--lr_backbone=1.0
--lr_decoder=1.0
--lr_embedding=1000.0
--lr_interpo=1.0
--lr_rot=1.0
--lr_ts=1.0
--mask_ratio=0.05
--min_loss=0
--new_query_num=500
--pad_num=100
--pad_radius=0.6
--pic_save=1
--pic_save_dir=sampling_eval
--query_num=1000
--query_radius=0.6
--r_con_w=1.0
--ratio=1.0
--ratio_source=Real
--recon_w=8.0
--refine_lr=0.0011
--refine_step=50
--regular_tps=0
--resume_base_model_name=model_69.pth
--roi_mask_pro=0.5
--roi_mask_r=3
--rot_1_w=1.0
--rot_2_w=1.0
--rot_regular=0.5
--scale_constant=64000.0
--scale_ratio=2.0
--size_w=8.0
--test_topk=100
--tran_w=8.0
--two_back=0
--use_adaptive_distance=0
--use_base=1
--use_consistency=0
--use_deform=0
--use_deform_emb=1
--use_dir_rpe=0
--use_distance=1
--use_fsnet=0
--use_inter=0
--use_interpo=1
--use_mean_init=0
--use_nocs_loss=1
--use_prob=1
--use_pso=0
--use_refine=1
--use_refine_loss=0
--use_sampling_test=0
--use_sampling_train=0
--lr_weight_model=1.0
--resume_gt_dir=/data_nvme/trained_models/pips_s1_512_v5
--resume_gt_model_name=model_69.pth
--s_topk=50
--stage=1.0
--use_stable_train=0
--nohelp
--nohelpfull
--nohelpshort
--nohelpxml
--noonly_check_args
--nopdb
--nopdb_post_mortem
--norun_with_pdb
--norun_with_profiling
--use_cprofile_for_profiling
--noalsologtostderr
--log_dir=
--logger_levels=
--nologtostderr
--showprefixforinfo
--stderrthreshold=fatal
--verbosity=0
--accumulate=1
--anneal_method=cosine
--anneal_point=0.5
--atten_factor=3.0
--backbone=gcn_equi4
--batch_size=4
--big_in_whole=1
--dataset=Real
--dataset_dir=/home/bingxing2/home/scx6md6/wby/nocs_data
--dataset_size=big
--debug_path=/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/debug.pt
--detection_dir=/data_nvme/nocs_data/segmentation_results
--device=cuda:0
--dim_list=128,128,256,256,512
--down_sample_method=farthest
--equi_neighbor_num=10
--eval_dataset=Real
--eval_out=output/pips_equi_v5
--fix_kernel=0
--gamma=0.1
--log_every=10
--lr=0.0003
--lr_decay_iters=50
--lr_pose=1.0
--lr_scheduler_name=flat_and_anneal
--model_save=output/pips_equi4_tovec3_no_global_pool
--nu_nocs=0
--num_workers=4
--obj_c=6
--optimizer_type=Ranger
--per_obj=mug
--pick_ratio=0.0
--poly_power=0.9
--qnet_config=qnet_equi_512.yaml
--qnet_version=v5
--random_points=1028
--resume=0
--resume_dir=/data_nvme/trained_models/pips_equi3_tovec3_no_global
--resume_model_name=model_49.pth
--resume_point=69
--rotation_noise_angle=20.0
--rotation_path=/home/bingxing2/home/scx6md6/wby/SOCS/nfmodel/rotation.pt
--sample_method=farthest
--save_every=5
--scale_invariant=0
--shuffle=1
--softmax=1
--total_epoch=70
--tovec_version=v3
--train=1
--train_steps=2000
--use_board=1
--use_global=0
--use_pick=0
--use_pool=0
--use_pred_scale=0
--verbose=0
--warmup_factor=0.001
--warmup_iters=1000
--warmup_method=linear
--weight_decay=0.0
--DZI_PAD_SCALE=1.5
--DZI_SCALE_RATIO=0.25
--DZI_SHIFT_RATIO=0.25
--DZI_TYPE=uniform
--R_c=3
--Ts_c=6
--align_origin=0
--aug_bb_pro=0.3
--aug_bc_pro=0.3
--aug_bg=1
--aug_deform=0
--aug_pc_pro=1.0
--aug_pc_r=0.001
--aug_rt_pro=1.0
--base_dir=/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/output/NFwhole_prior_7
--bin_size=100
--cd_w=0.0
--consistency_w=1.0
--cur_cat_model_num=0
--cur_eval_index=-1
--cut_backbone=0
--debug=0
--deform_w=10.0
--eval_inference_only=0
--eval_optimizer=adam
--eval_use_deform=0
--eval_visualize_pcl=0
--face_recon_c=30
--fake_grid_num=500
--fake_radius=0.6
--feat_c_R=1280
--feat_c_ts=1283
--feat_face=768
--feat_for_ts=0
--force_sym=0
--fsnet_loss_type=l1
--gcn_n_num=10
--gcn_sup_num=7
--grad_w=1.0
--img_size=256
--inter_R_lr=1.0
--inter_nocs_lr=1.0
--inter_t_lr=1.0
--interpo_w=10.0
--keypoint_num=36
--keypoint_path=/home/bingxing2/home/scx6md6/wby/nocs_data/keypoint.pkl
--kld_w=0.0
--lr_backbone=1.0
--lr_decoder=1.0
--lr_embedding=1000.0
--lr_interpo=1.0
--lr_rot=1.0
--lr_ts=1.0
--mask_ratio=0.05
--min_loss=0
--new_query_num=500
--pad_num=100
--pad_radius=0.6
--pic_save=1
--pic_save_dir=sampling_eval
--query_num=1000
--query_radius=0.6
--r_con_w=1.0
--ratio=1.0
--ratio_source=Real
--recon_w=8.0
--refine_lr=0.0011
--refine_step=50
--regular_tps=0
--resume_base_model_name=model_69.pth
--roi_mask_pro=0.5
--roi_mask_r=3
--rot_1_w=1.0
--rot_2_w=1.0
--rot_regular=0.5
--scale_constant=64000.0
--scale_ratio=2.0
--size_w=8.0
--test_topk=100
--tran_w=8.0
--two_back=0
--use_adaptive_distance=0
--use_base=1
--use_consistency=0
--use_deform=0
--use_deform_emb=1
--use_dir_rpe=0
--use_distance=1
--use_fsnet=0
--use_inter=0
--use_interpo=1
--use_mean_init=0
--use_nocs_loss=1
--use_prob=1
--use_pso=0
--use_refine=1
--use_refine_loss=0
--use_sampling_test=0
--use_sampling_train=0
--lr_weight_model=1.0
--resume_gt_dir=/data_nvme/trained_models/pips_s1_512_v5
--resume_gt_model_name=model_69.pth
--s_topk=50
--stage=1.0
--use_stable_train=0
--nohelp
--nohelpfull
--nohelpshort
--nohelpxml
--noonly_check_args
--nopdb
--nopdb_post_mortem
--norun_with_pdb
--norun_with_profiling
--use_cprofile_for_profiling
--noalsologtostderr
--log_dir=
--logger_levels=
--nologtostderr
--showprefixforinfo
--stderrthreshold=fatal
--verbosity=0
--accumulate=1
--anneal_method=cosine
--anneal_point=0.5
--atten_factor=3.0
--backbone=gcn_equi4
--batch_size=4
--big_in_whole=1
--dataset=Real
--dataset_dir=/home/bingxing2/home/scx6md6/wby/nocs_data
--dataset_size=big
--debug_path=/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/debug.pt
--detection_dir=/data_nvme/nocs_data/segmentation_results
--device=cuda:0
--dim_list=128,128,256,256,512
--down_sample_method=farthest
--equi_neighbor_num=10
--eval_dataset=Real
--eval_out=output/pips_equi_v5
--fix_kernel=0
--gamma=0.1
--log_every=10
--lr=0.0003
--lr_decay_iters=50
--lr_pose=1.0
--lr_scheduler_name=flat_and_anneal
--model_save=output/pips_equi4_tovec3_no_global_pool
--nu_nocs=0
--num_workers=4
--obj_c=6
--optimizer_type=Ranger
--per_obj=mug
--pick_ratio=0.0
--poly_power=0.9
--qnet_config=qnet_equi_512.yaml
--qnet_version=v5
--random_points=1028
--resume=0
--resume_dir=/data_nvme/trained_models/pips_equi3_tovec3_no_global
--resume_model_name=model_49.pth
--resume_point=69
--rotation_noise_angle=20.0
--rotation_path=/home/bingxing2/home/scx6md6/wby/SOCS/nfmodel/rotation.pt
--sample_method=farthest
--save_every=5
--scale_invariant=0
--shuffle=1
--softmax=1
--total_epoch=70
--tovec_version=v3
--train=1
--train_steps=2000
--use_board=1
--use_global=0
--use_pick=0
--use_pool=0
--use_pred_scale=0
--verbose=0
--warmup_factor=0.001
--warmup_iters=1000
--warmup_method=linear
--weight_decay=0.0
--DZI_PAD_SCALE=1.5
--DZI_SCALE_RATIO=0.25
--DZI_SHIFT_RATIO=0.25
--DZI_TYPE=uniform
--R_c=3
--Ts_c=6
--align_origin=0
--aug_bb_pro=0.3
--aug_bc_pro=0.3
--aug_bg=1
--aug_deform=0
--aug_pc_pro=1.0
--aug_pc_r=0.001
--aug_rt_pro=1.0
--base_dir=/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/output/NFwhole_prior_7
--bin_size=100
--cd_w=0.0
--consistency_w=1.0
--cur_cat_model_num=0
--cur_eval_index=-1
--cut_backbone=0
--debug=0
--deform_w=10.0
--eval_inference_only=0
--eval_optimizer=adam
--eval_use_deform=0
--eval_visualize_pcl=0
--face_recon_c=30
--fake_grid_num=500
--fake_radius=0.6
--feat_c_R=1280
--feat_c_ts=1283
--feat_face=768
--feat_for_ts=0
--force_sym=0
--fsnet_loss_type=l1
--gcn_n_num=10
--gcn_sup_num=7
--grad_w=1.0
--img_size=256
--inter_R_lr=1.0
--inter_nocs_lr=1.0
--inter_t_lr=1.0
--interpo_w=10.0
--keypoint_num=36
--keypoint_path=/home/bingxing2/home/scx6md6/wby/nocs_data/keypoint.pkl
--kld_w=0.0
--lr_backbone=1.0
--lr_decoder=1.0
--lr_embedding=1000.0
--lr_interpo=1.0
--lr_rot=1.0
--lr_ts=1.0
--mask_ratio=0.05
--min_loss=0
--new_query_num=500
--pad_num=100
--pad_radius=0.6
--pic_save=1
--pic_save_dir=sampling_eval
--query_num=1000
--query_radius=0.6
--r_con_w=1.0
--ratio=1.0
--ratio_source=Real
--recon_w=8.0
--refine_lr=0.0011
--refine_step=50
--regular_tps=0
--resume_base_model_name=model_69.pth
--roi_mask_pro=0.5
--roi_mask_r=3
--rot_1_w=1.0
--rot_2_w=1.0
--rot_regular=0.5
--scale_constant=64000.0
--scale_ratio=2.0
--size_w=8.0
--test_topk=100
--tran_w=8.0
--two_back=0
--use_adaptive_distance=0
--use_base=1
--use_consistency=0
--use_deform=0
--use_deform_emb=1
--use_dir_rpe=0
--use_distance=1
--use_fsnet=0
--use_inter=0
--use_interpo=1
--use_mean_init=0
--use_nocs_loss=1
--use_prob=1
--use_pso=0
--use_refine=1
--use_refine_loss=0
--use_sampling_test=0
--use_sampling_train=0
--lr_weight_model=1.0
--resume_gt_dir=/data_nvme/trained_models/pips_s1_512_v5
--resume_gt_model_name=model_69.pth
--s_topk=50
--stage=1.0
--use_stable_train=0
--nohelp
--nohelpfull
--nohelpshort
--nohelpxml
--noonly_check_args
--nopdb
--nopdb_post_mortem
--norun_with_pdb
--norun_with_profiling
--use_cprofile_for_profiling
--noalsologtostderr
--log_dir=
--logger_levels=
--nologtostderr
--showprefixforinfo
--stderrthreshold=fatal
--verbosity=0
--accumulate=1
--anneal_method=cosine
--anneal_point=0.5
--atten_factor=3.0
--backbone=gcn_equi4
--batch_size=4
--big_in_whole=1
--dataset=Real
--dataset_dir=/home/bingxing2/home/scx6md6/wby/nocs_data
--dataset_size=big
--debug_path=/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/debug.pt
--detection_dir=/data_nvme/nocs_data/segmentation_results
--device=cuda:0
--dim_list=128,128,256,256,512
--down_sample_method=farthest
--equi_neighbor_num=10
--eval_dataset=Real
--eval_out=output/pips_equi_v5
--fix_kernel=0
--gamma=0.1
--log_every=10
--lr=0.0003
--lr_decay_iters=50
--lr_pose=1.0
--lr_scheduler_name=flat_and_anneal
--model_save=output/pips_equi4_tovec3_no_global_pool
--nu_nocs=0
--num_workers=4
--obj_c=6
--optimizer_type=Ranger
--per_obj=mug
--pick_ratio=0.0
--poly_power=0.9
--qnet_config=qnet_equi_512.yaml
--qnet_version=v5
--random_points=1028
--resume=0
--resume_dir=/data_nvme/trained_models/pips_equi3_tovec3_no_global
--resume_model_name=model_49.pth
--resume_point=69
--rotation_noise_angle=20.0
--rotation_path=/home/bingxing2/home/scx6md6/wby/SOCS/nfmodel/rotation.pt
--sample_method=farthest
--save_every=5
--scale_invariant=0
--shuffle=1
--softmax=1
--total_epoch=70
--tovec_version=v3
--train=1
--train_steps=2000
--use_board=1
--use_global=0
--use_pick=0
--use_pool=0
--use_pred_scale=0
--verbose=0
--warmup_factor=0.001
--warmup_iters=1000
--warmup_method=linear
--weight_decay=0.0
--DZI_PAD_SCALE=1.5
--DZI_SCALE_RATIO=0.25
--DZI_SHIFT_RATIO=0.25
--DZI_TYPE=uniform
--R_c=3
--Ts_c=6
--align_origin=0
--aug_bb_pro=0.3
--aug_bc_pro=0.3
--aug_bg=1
--aug_deform=0
--aug_pc_pro=1.0
--aug_pc_r=0.001
--aug_rt_pro=1.0
--base_dir=/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/output/NFwhole_prior_7
--bin_size=100
--cd_w=0.0
--consistency_w=1.0
--cur_cat_model_num=0
--cur_eval_index=-1
--cut_backbone=0
--debug=0
--deform_w=10.0
--eval_inference_only=0
--eval_optimizer=adam
--eval_use_deform=0
--eval_visualize_pcl=0
--face_recon_c=30
--fake_grid_num=500
--fake_radius=0.6
--feat_c_R=1280
--feat_c_ts=1283
--feat_face=768
--feat_for_ts=0
--force_sym=0
--fsnet_loss_type=l1
--gcn_n_num=10
--gcn_sup_num=7
--grad_w=1.0
--img_size=256
--inter_R_lr=1.0
--inter_nocs_lr=1.0
--inter_t_lr=1.0
--interpo_w=10.0
--keypoint_num=36
--keypoint_path=/home/bingxing2/home/scx6md6/wby/nocs_data/keypoint.pkl
--kld_w=0.0
--lr_backbone=1.0
--lr_decoder=1.0
--lr_embedding=1000.0
--lr_interpo=1.0
--lr_rot=1.0
--lr_ts=1.0
--mask_ratio=0.05
--min_loss=0
--new_query_num=500
--pad_num=100
--pad_radius=0.6
--pic_save=1
--pic_save_dir=sampling_eval
--query_num=1000
--query_radius=0.6
--r_con_w=1.0
--ratio=1.0
--ratio_source=Real
--recon_w=8.0
--refine_lr=0.0011
--refine_step=50
--regular_tps=0
--resume_base_model_name=model_69.pth
--roi_mask_pro=0.5
--roi_mask_r=3
--rot_1_w=1.0
--rot_2_w=1.0
--rot_regular=0.5
--scale_constant=64000.0
--scale_ratio=2.0
--size_w=8.0
--test_topk=100
--tran_w=8.0
--two_back=0
--use_adaptive_distance=0
--use_base=1
--use_consistency=0
--use_deform=0
--use_deform_emb=1
--use_dir_rpe=0
--use_distance=1
--use_fsnet=0
--use_inter=0
--use_interpo=1
--use_mean_init=0
--use_nocs_loss=1
--use_prob=1
--use_pso=0
--use_refine=1
--use_refine_loss=0
--use_sampling_test=0
--use_sampling_train=0
--lr_weight_model=1.0
--resume_gt_dir=/data_nvme/trained_models/pips_s1_512_v5
--resume_gt_model_name=model_69.pth
--s_topk=50
--stage=1.0
--use_stable_train=0


config/lmo/config.py
from __future__ import print_function

import absl.flags as flags
import config.lmo.pips_s_config
import config.lmo.fix_config
import numpy as np


np.bool=np.bool_
np.float=np.float_
np.int=np.int_
# eval
flags.DEFINE_string('eval_out', 'lmo', '')

flags.DEFINE_float('stage',2,'')

flags.DEFINE_string('pic_save_dir', 'lmo_eval_per_cat', '')

flags.DEFINE_integer('verbose',0 ,'')
flags.DEFINE_integer('shuffle',1, '')
flags.DEFINE_integer('use_board',1, '')



flags.DEFINE_integer('use_global', 1, '')

flags.DEFINE_list('obj_list',[1],'')
# train
flags.DEFINE_string('rotation_path','/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/rotation.pt', '')
flags.DEFINE_string('debug_path','/home/wanboyan/Documents/cmr-master/extern/GPV_Pose-master/debug.pt', '')








flags.DEFINE_string('backbone','gcn_equi3', '')
# flags.DEFINE_string('backbone','gcn', '')


flags.DEFINE_string('qnet_version','v5','')
# flags.DEFINE_string('qnet_version','v6','')


# flags.DEFINE_string('qnet_config','qnet_equi_128.yaml', '')
flags.DEFINE_string('qnet_config','qnet_equi_512.yaml', '')
# flags.DEFINE_string('qnet_config','qnet.yaml', '')


flags.DEFINE_list('dim_list',[128,128,256,256,512],'')
# flags.DEFINE_list('dim_list',[32,32,64,64,128],'')

#

flags.DEFINE_string('dataset_dir', '/data_sata/lmo', 'path to the dataset')





# point selection



flags.DEFINE_string('device', 'cuda:0', '')
flags.DEFINE_integer("num_workers", 4, "cpu cores for loading dataset")
flags.DEFINE_integer('batch_size',4, '')
flags.DEFINE_integer('total_epoch', 30, 'total epoches in training')
flags.DEFINE_integer('train_steps', 2000, 'number of batches in each epoch')  # batchsize is 8, then 3000
#####################space is not enough, trade time for space####################
flags.DEFINE_integer('accumulate', 1, '')   # the real batch size is batchsize x accumulate




flags.DEFINE_float('lr', 3e-4, '')
flags.DEFINE_integer('lr_decay_iters', 50, '')  # some parameter for the scheduler### optimizer  ####
flags.DEFINE_string('lr_scheduler_name', 'flat_and_anneal', 'linear/warm_flat_anneal/')
flags.DEFINE_string('anneal_method', 'cosine', '')
flags.DEFINE_float('anneal_point', 0.5, '')
flags.DEFINE_string('optimizer_type', 'Ranger', '')
flags.DEFINE_float('weight_decay', 0.0, '')
flags.DEFINE_float('warmup_factor', 0.001, '')
flags.DEFINE_integer('warmup_iters', 1000, '')
flags.DEFINE_string('warmup_method', 'linear', '')
flags.DEFINE_float('gamma', 0.1, '')
flags.DEFINE_float('poly_power', 0.9, '')

# save parameters
#model to save
flags.DEFINE_integer('save_every', 5, '')  # save models every 'save_every' epoch
flags.DEFINE_integer('log_every', 10, '')  # save log file every 100 iterations
flags.DEFINE_string('model_save', 'output/lmo', 'path to save checkpoint')


# resume
flags.DEFINE_integer('resume',0, '1 for resume, 0 for training from the start')
flags.DEFINE_string('resume_dir', '/data_nvme/trained_models/equi_lmo','')
flags.DEFINE_string('resume_model_name', 'model_64.pth','')

flags.DEFINE_integer('resume_point', 19, 'the epoch to continue the training')






